lib/Test/Nginx/Util.pm

```diff
@@ -243,7 +243,28 @@ our $ForceRestartOnTest     = (defined $ENV{TEST_NGINX_FORCE_RESTART_ON_TEST})
 
 if ($Randomize) {
     srand $$;
-    $ServerPort = int(rand 60000) + 1025;
+
+    my $i = 0;
+    while ($i < 1000) {
+        $ServerPort = int(rand 60000) + 1025;
+
+        my $sock = IO::Socket::INET->new(
+            PeerAddr => $ServerAddr,
+            PeerPort => $ServerPort,
+            Proto => 'tcp',
+            Timeout => 0.1,
+        );
+
+        last if !$sock;
+
+        $sock->close();
+
+        if ($Verbose) {
+            warn "try again, port: $ServerPort is using!";
+        }
+        $i++;
+    }
+
```

Comment:
We should abort here if we do not find any ports usable instead of blindly proceed.